include/TimePos.h

```diff
@@ -107,7 +110,7 @@ class LMMS_EXPORT TimePos
 	static double ticksToMilliseconds( double ticks, bpm_t beatsPerMinute );
 
 private:
-	tick_t m_ticks;
+	std::atomic<tick_t> m_ticks;
```

Comment:
Would wrapping a ``TimePos`` in a ``std::atomic`` be lock-free on most architectures (mainly x86_64)? The standard only guarantees ``std::atomic_flag`` will be lock-free, so I've been wondering what we can safely assume will be lock-free for some time.

Reply:
I would only expect it to depend on the size of the object, and since the only non-static member variable of `TimePos` is `m_ticks`, there shouldn't be any difference as to whether they require locks. You can also assert it to be sure - `static_assert(std::atomic<TimePos>::is_always_lock_free);`.

Reply:
Did some digging, and found out that ``Song::PlayPos``, which inherits from ``TimePos``, uses up 24 bytes, where the largest atomic instruction generally supports only up to 16 bytes. My changes seemed to be focused on fixing data races from ``Song::PlayPos``, but I can't make any forward progress unless I change that class somehow.

Reply:
I tried making ``Song::PlayPos`` a certain size so it can work with ``std::atomic``, but it would require a large change to the codebase that doesn't seem worth it, and even if everything compiles with the changes, the linker complains about undefined references to the atomic functions. I wanted to fix the data race that occurred from me moving the playhead in the Song Editor, thus involving ``Song::PlayPos``, but I might have to leave this to be addressed another time. Users of the basic ``TimePos`` that are racing with the audio threads however, could use ``std::atomic``, but that can be done in a different PR. I will revert this change though.